icecreamshop/src/main/java/com/summerschool/icecreamshop/service/BasketService.java

```diff
@@ -0,0 +1,27 @@
+package com.summerschool.icecreamshop.service;
+
+import com.summerschool.icecreamshop.model.Basket;
+import com.summerschool.icecreamshop.model.BasketProduct;
+import com.summerschool.icecreamshop.repository.BasketRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class BasketService {
+
+    @Autowired
+    private BasketRepository basketRepository;
+
+    public Optional<Basket> get (Long id){
+        return basketRepository.findById(id);
+    }
+
+    public Basket update (Basket basket , List<BasketProduct> l){
```

Comment:
Poti sa redenumesti te rog al doilea parametru in `basketProductList` sau ceva mai explicit decat notatia folosita acum.